inversify/inversify.d.ts

```diff
@@ -1,141 +1,273 @@
+            toConstantValue(value: T): BindingWhenOnSyntax<T>;
+            toDynamicValue(func: () => T): BindingWhenOnSyntax<T>;
+            toConstructor<T2>(constructor: Newable<T2>): BindingWhenOnSyntax<T>;
+            toFactory<T2>(factory: FactoryCreator<T2>): BindingWhenOnSyntax<T>;
+            toFunction(func: T): BindingWhenOnSyntax<T>;
+            toAutoFactory<T2>(serviceIdentifier: (string|Symbol|Newable<T2>)): BindingWhenOnSyntax<T>;
+            toProvider<T2>(provider: ProviderCreator<T2>): BindingWhenOnSyntax<T>;
+        }
+
+        export interface BindingWhenOnSyntax<T> extends BindingWhenSyntax<T>, BindingOnSyntax<T> {}
+
+        export interface BindingWhenSyntax<T> {
+            when(constraint: (request: Request) => boolean): BindingOnSyntax<T>;
+            whenTargetNamed(name: string): BindingOnSyntax<T>;
+            whenTargetTagged(tag: string, value: any): BindingOnSyntax<T>;
+            whenInjectedInto(parent: (Function|string)): BindingOnSyntax<T>;
+            whenParentNamed(name: string): BindingOnSyntax<T>;
+            whenParentTagged(tag: string, value: any): BindingOnSyntax<T>;
+            whenAnyAncestorIs(ancestor: (Function|string)): BindingOnSyntax<T>;
+            whenNoAncestorIs(ancestor: (Function|string)): BindingOnSyntax<T>;
+            whenAnyAncestorNamed(name: string): BindingOnSyntax<T>;
+            whenAnyAncestorTagged(tag: string, value: any): BindingOnSyntax<T>;
+            whenNoAncestorNamed(name: string): BindingOnSyntax<T>;
+            whenNoAncestorTagged(tag: string, value: any): BindingOnSyntax<T>;
+            whenAnyAncestorMatches(constraint: (request: Request) => boolean): BindingOnSyntax<T>;
+            whenNoAncestorMatches(constraint: (request: Request) => boolean): BindingOnSyntax<T>;
+        }
 
-    export interface IMetadata {
-        key: string;
-        value: any;
     }
 
-    export var Kernel: IKernelConstructor;
-    export var decorate: any;
-    export function inject(...typeIdentifiers: string[]): (typeConstructor: any) => void;
-    export var tagged: any;
-    export var named: any;
-    export var paramNames: any;
+    export var Kernel: interfaces.KernelConstructor;
+    export var KernelModule: interfaces.KernelModuleConstructor;
+    export var decorate: (decorator: (ClassDecorator|ParameterDecorator), target: any, parameterIndex?: number) => void;
+    export function injectable(): (typeConstructor: any) => void;
+    export function tagged(metadataKey: string, metadataValue: any): (target: any, targetKey: string, index?: number) => any;
+    export function named(name: string): (target: any, targetKey: string, index?: number) => any;
+    export function targetName(name: string): (target: any, targetKey: string, index: number) => any;
+    export function inject(serviceIdentifier: (string|Symbol|any)): (target: any, targetKey: string, index?: number) => any;
```

Comment:
this does not look right. `any|...` is just `any`. there is no value of adding the two other types.